src/seedu/addressbook/data/person/Address.java

```diff
@@ -30,14 +27,13 @@
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        
-        this.value = address;
+        super.isPrivate = isPrivate; 
+        super.value = address;
         splitAddress(address);
```

Comment:
Why not just have `super(isPrivate, value)` You can have your validations done in your super class as well.